src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/services/service-form/service-form.component.ts

```diff
@@ -420,7 +438,58 @@ export class ServiceFormComponent extends CdForm implements OnInit {
         ]
       ],
       grafana_port: [null, [CdValidators.number(false)]],
-      grafana_admin_password: [null]
+      grafana_admin_password: [null],
+      // oauth2-proxy
+      provider_display_name: [
+        'My OIDC provider',
+        [
+          CdValidators.requiredIf({
+            service_type: 'oauth2-proxy'
+          })
+        ]
+      ],
+      client_id: [
+        null,
+        [
+          CdValidators.requiredIf({
+            service_type: 'oauth2-proxy'
+          })
+        ]
+      ],
+      client_secret: [
+        null,
+        [
+          CdValidators.requiredIf({
+            service_type: 'oauth2-proxy'
+          })
+        ]
+      ],
+      oidc_issuer_url: [
+        null,
+        [
+          CdValidators.requiredIf({
+            service_type: 'oauth2-proxy'
+          })
+        ]
+      ],
+      https_address: [
+        null,
+        [
+          CdValidators.custom('httpsAddressPattern', (value: string) => {
+            if (_.isEmpty(value)) {
+              return false;
```

Comment:
Lets create a separate function for this `oauthAddressTest()` .

Reply:
Don't really see the appeal for it, tbh

Reply:
The reason is to maintain code readability and maintenance.

Reply:
Added the function